src/decays/one_loop_decay_diagrams.cpp

```diff
@@ -0,0 +1,7177 @@
+
+   return result;
+}
+
+/**
+ * @brief Evaluates T5G4N16 diagram for process S -> FF
+ *
+ * @param[in] mext1 mass of external field S[1]
+ * @param[in] mext2 mass of external field F[2]
+ * @param[in] mext3 mass of external field F[3]
+ * @param[in] mIS4 mass of internal field S[4]
+ * @param[in] mLS5 mass of internal field S[5]
+ * @param[in] mLV6 mass of internal field V[6]
+ * @param[in] CpF2F3cS4PL coupling Cp[F[2], F[3], -S[4]][PL]
+ * @param[in] CpF2F3cS4PR coupling Cp[F[2], F[3], -S[4]][PR]
+ * @param[in] CpS1cS5cV6 coupling Cp[S[1], -S[5], -V[6]][Mom[S[1]] - Mom[-S[5]]
+    ]
+ * @param[in] CpS4S5V6 coupling Cp[S[4], S[5], V[6]][Mom[S[4]] - Mom[S[5]]]
+ *
+ * @return value of the one-loop diagram
+ */
+Decay_amplitude_SFF calculate_diagram_SFF_t5g4n16_SSV(
+   double mext1, double mext2, double mext3,
+   double mIS4, double mLS5, double mLV6,
+   const std::complex<double>& CpF2F3cS4PL, const std::complex<double>&
+      CpF2F3cS4PR, const std::complex<double>& CpS1cS5cV6, const std::complex<
+      double>& CpS4S5V6,
+   double scale)
+{
+   auto& lib = Loop_library::get();
+
+   const auto a0tmp1 = lib.A0(mLV6*mLV6, scale*scale);
+   const auto b0tmp2 = lib.B0(mext1*mext1, mLS5*mLS5, mLV6*mLV6,
+      scale*scale);
+   const auto b1tmp3 = lib.B1(mext1*mext1, mLS5*mLS5, mLV6*mLV6,
+      scale*scale);
+
+   Decay_amplitude_SFF result;
+
+   result.m_decay = mext1;
+   result.m_fermion_1 = mext2;
+   result.m_fermion_2 = mext3;
+
+   result.form_factor_left = oneOver16PiSqr*(std::complex<double>(0,1)*
+      CpF2F3cS4PL*CpS1cS5cV6*CpS4S5V6*Den(Sqr(mext1),Sqr(mIS4))*(a0tmp1 - 2*
+      b1tmp3*Sqr(mext1) + b0tmp2*(Sqr(mext1) + Sqr(mLS5))));
+   result.form_factor_right = oneOver16PiSqr*(std::complex<double>(0,1)*
+      CpF2F3cS4PR*CpS1cS5cV6*CpS4S5V6*Den(Sqr(mext1),Sqr(mIS4))*(a0tmp1 - 2*
+      b1tmp3*Sqr(mext1) + b0tmp2*(Sqr(mext1) + Sqr(mLS5))));
+
+   return result;
+}
+
+/**
+ * @brief Evaluates T5G5N17 diagram for process S -> FF
+ *
+ * @param[in] mext1 mass of external field S[1]
+ * @param[in] mext2 mass of external field F[2]
+ * @param[in] mext3 mass of external field F[3]
+ * @param[in] mIS4 mass of internal field S[4]
+ * @param[in] mLV5 mass of internal field V[5]
+ * @param[in] mLV6 mass of internal field V[6]
+ * @param[in] CpF2F3cS4PL coupling Cp[F[2], F[3], -S[4]][PL]
+ * @param[in] CpF2F3cS4PR coupling Cp[F[2], F[3], -S[4]][PR]
+ * @param[in] CpS1cV5cV6 coupling Cp[S[1], -V[5], -V[6]][g[lt2, lt3]]
+ * @param[in] CpS4V5V6 coupling Cp[S[4], V[5], V[6]][g[lt2, lt3]]
+ *
+ * @return value of the one-loop diagram
+ */
+Decay_amplitude_SFF calculate_diagram_SFF_t5g5n17_SVV(
+   double mext1, double mext2, double mext3,
+   double mIS4, double mLV5, double mLV6,
+   const std::complex<double>& CpF2F3cS4PL, const std::complex<double>&
+      CpF2F3cS4PR, const std::complex<double>& CpS1cV5cV6, const std::complex<
+      double>& CpS4V5V6,
+   double scale, double finite)
+
```

Comment:
This file is huuuge. I can't say I have reviewed it properly :(.

Reply:
Why thank you ;) This is auto-generated. i don't expect you to look through all of it.